src/coord/coord.rs

```diff
@@ -478,16 +489,31 @@ where
 
                 Message::StatementReady {
                     conn_id,
-                    mut session,
+                    session,
                     tx,
                     result,
                     params,
-                } => {
-                    let result = result
-                        .and_then(|stmt| self.handle_statement(&session, stmt, &params))
-                        .and_then(|plan| self.sequence_plan(&mut session, plan, conn_id));
-                    let _ = tx.send(Response { result, session });
-                }
+                } => match result.and_then(|stmt| self.handle_statement(&session, stmt, &params)) {
+                    Ok(plan) => self.sequence_plan(&internal_cmd_tx, tx, session, plan, conn_id),
+                    Err(e) => tx.send(Err(e), session),
+                },
+
+                Message::SinkConnectorReady {
+                    session,
+                    tx,
+                    id,
+                    result,
+                } => match result {
+                    Ok(connector) => {
+                        // NOTE: we must not fail from here on out. We have a
+                        // connector, which means there is external state (like
+                        // a Kafka topic) that's been created on our behalf. If
+                        // we fail now, we'll leak that external state.
+                        self.handle_sink_connector_ready(id, connector);
+                        tx.send(Ok(ExecuteResponse::CreatedSink { existed: false }), session);
+                    }
+                    Err(e) => tx.send(Err(e), session),
```

Comment:
I think we need to get rid of the previously reserved name here?

Reply:
Yes, thank you!

Reply:
Ok, done.